packages/@glimmer/runtime/lib/compiled/expressions/concat.ts

```diff
@@ -5,10 +5,10 @@ import { enumerate } from '@glimmer/util';
 
 export function createConcatRef(partsRefs: Reference[]) {
   return createComputeRef(() => {
-    let parts = new Array<string>();
+    const parts = new Array<string>(partsRefs.length).fill('');
 
     for (const [i, ref] of enumerate(partsRefs)) {
-      let value = valueForRef(ref) as Maybe<Dict>;
+      const value = valueForRef(ref) as Maybe<Dict>;
 
       if (value !== null && value !== undefined) {
         parts[i] = castToString(value);
```

Comment:
using `Array.fill` on line 8 we avoiding holes in `parts` array

Reply:
@chancancode if value will be `null` or `undefined` - hole is created for this index in array

Reply:
I see, yes that is definitely a problem!

Reply:
The problem is in skipping the assignment conditionally though, not necessary with the `[i] = ` style of filling an array

Reply:
@chancancode should it be `array.push` in this case?

Reply:
The existing for loop should be fine/equivalent `.push` as long as we don't skip it conditionally, I would think

Reply:
Ah, but yeah, I see, if we switch back to not pre-allocating the length, then we can just `.push` as many times as as needed and then do the `.join`, that seems good to me. That can also get rid of the `enumerate()` thing here I assume

Reply:
done